pandas/core/internals/blocks.py

```diff
@@ -1297,24 +1296,20 @@ def take_nd(self, indexer, axis, new_mgr_locs=None, fill_tuple=None):
 
         if fill_tuple is None:
             fill_value = self.fill_value
-            new_values = algos.take_nd(
-                values, indexer, axis=axis, allow_fill=False, fill_value=fill_value
-            )
+            allow_fill = False
         else:
             fill_value = fill_tuple[0]
-            new_values = algos.take_nd(
-                values, indexer, axis=axis, allow_fill=True, fill_value=fill_value
-            )
+            allow_fill = True
```

Comment:
In the ExtensionBlock.take_nd method we have this same if/elif for fill_tuple/fill_value, but we don't set allow_fill there.  Seems fishy.  @jorisvandenbossche any idea if this is intentional?

Reply:
Do you know for what case the `allow_fill=False` branch gets hit?

In the end, I think this is only an optimization for when you know that there are no -1's in the indexer. Because in the internals, I think we never use the numpy-like indexing semantics of -1 meaning the last element.